docs/pbc/all/multi-factor-auth/202505.0/multi-factor-auth.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Multi-Factor Authentication
+description: Learn about Multi Factor Authentication (MFA) that you can use within your Spryker project.
+template: concept-topic-template
+last_updated: Mar 05, 2026
+---
+
+{% info_block warningBox "⚠️ WARNING: BETA FEATURE" %}
+
+**THE MULTI-FACTOR AUTH MODULE IS CURRENTLY IN BETA.**
+
+This feature is not yet considered stable and **may be significantly changed in future releases**.
+
+By using this feature, you accept full responsibility for any potential issues, including breaking changes, limited support, and incomplete functionality.
+
+**Use in production environments is not recommended** unless you are prepared to adapt to future updates.
+
+{% endinfo_block %}
+
+Multi-Factor Authentication adds an extra layer of security to your account.
+It allows customers to enhance their account security by requiring additional verification steps beyond the standard username and password.
+
+## Benefits of MFA
+
+- **Enhanced security**: MFA adds an extra layer of security to your account, making it harder for attackers to gain access.
+- **Enhanced privacy**: MFA helps protect your privacy by requiring additional verification steps beyond the standard username and password.
+- **Reduced risk**: MFA reduces the risk of unauthorized access, as it requires multiple factors to be verified before allowing access.
+- **Compliance**: MFA helps you comply with security regulations and standards.
+
+## Flow Overview
+
+1) **Initial MFA check**: The system first verifies whether any MFA types are enabled for the customer. 
+If no MFA methods are enabled, the authentication process continues without additional validation.
+
+2) **Fetch enabled MFA types**: If MFA is enabled, the system sends a request to `multi-factor-auth/get-customer-enabled-types` to retrieve a list of enabled MFA types for the customer.
+
+3) **Evaluate the number of enabled MFA types**: If more than one MFA method is enabled, the system presents a selection screen where the user must choose their preferred authentication method.
+If only one MFA method is enabled, the system proceeds to the next step automatically.
+
+4) **Send the authentication code**: If only one MFA method is enabled, the system automatically sends the authentication code using the appropriate method via `multi-factor-auth/send-customer-code`.
+If multiple methods were available and the user has selected one, the authentication code is sent through the selected method.
+
+5) **Code validation**: After the authentication code is sent, the system presents a code validation form to the user. The user enters the received authentication code in the form. 
+If the code is correct, authentication is successful. If incorrect, the user must retry.
```

Comment:
Does the user need to generate a new code or double-check the code and try entering again?

Reply:
Double-check the code and enter it again. So far the customer has 3 attempts (it's described below) only after that the page is refreshed and the flow should be passed again with a newly generated code